mesop/web/src/component_renderer/component_renderer.ts

```diff
@@ -143,6 +156,30 @@ export class ComponentRenderer {
   }
 
   ngOnChanges() {
+    if (this.customElement) {
+      // This is a naive way to apply changes by removing all the children
+      // and creating new children. In the future, this can be optimized
+      // to be more performant, but this naive approach should have the
+      // correct behavior, albeit inefficiently.
+
+      // Clear existing children
+      for (const element of Array.from(
+        this.customElement.querySelectorAll(COMPONENT_RENDERER_ELEMENT_NAME),
+      )) {
+        this.customElement.removeChild(element);
+      }
+
+      // Update the custom element and its children
+      this.updateCustomElement(this.customElement);
+      for (const child of this.component.getChildrenList()) {
```

Comment:
[question] Meant to ask this earlier but can we include core components within a web component with a slot? I know the examples show that we can do it with web components.

Reply:
Yep, you can use built-in components inside a web component slot. I've updated the example.